Issue #123: Show an alert when there are pending transactions

alexandermushinskiy opened the pull request:
Fixes #123 

Changes:
- show notification bar if there is at least one pending transaction.

Senyoret1 commented:
https://github.com/skycoin/skycoin-web/issues/122 has been solved. Now the balance is updated using this api endpoint: https://github.com/skycoin/skycoin/blob/develop/src/api/README.md#get-balance-of-addresses. The relevant part is that this endpoint returns two balances:

- confirmed: the current confirmed balance of all the addresses.
- predicted: the balance that the addresses will have after the pending transactions are processed.

This means that, if the current balance and the predicted balance do not match, then there are pending transactions affecting one or more of the user's addresses. It is important to keep in mind that it is necessary to check both the coins and the hours, if any of the two values does not match, then there are pending transactions. Knowing this, it is no longer necessary to call `/pendingTxs`, which simplifies the code and saves bandwidth.

alexandermushinskiy commented:
In the last commit:
- merged with master;
- updated logic to define existing pending transaction using balance properties confirmed and predicted;
- removed method which returns observable pending transactions to make it consistence with whole code;
- added hasPendingTransactions (ReplaySubject<boolean>) instead of pendingTransaction (ReplaySubject<any[]>);
- set update time to 20 seconds instead of 20 minutes.

Senyoret1 commented on src/app/services/wallet.service.ts:
```diff
@@ -311,17 +316,21 @@ export class WalletService {
   private calculateTimeSinceLastUpdate() {
     const diffMs: number = this.lastBalancesUpdateTime.getTime() - new Date().getTime();
-    const timeSinceLastUpdate = Math.abs(Math.round((diffMs / 1000 / 60)));
+    const timeSinceLastUpdate = this.convertDecimalToInt(diffMs / 1000);
```
`diffMs` and `timeSinceLastUpdate` must be inside `this._ngZone.run(() => {`, otherwise the updating is not done correctly.

alexandermushinskiy replied:
The comment has been fixed.
